hilti/toolchain/src/compiler/global-optimizer.cc

```diff
@@ -87,10 +93,68 @@ struct Visitor : hilti::visitor::PreOrder<bool, Visitor> {
         return std::make_tuple(ns_ns, ns_local, local);
     }
 
+    static std::optional<GlobalOptimizer::ID> getID(const type::struct_::Field& x, position_t p) {
+        auto field_id = x.id();
+
+        auto struct_type = typeID(p.parent().as<type::Struct>());
+        if ( ! struct_type )
+            return {};
+
+        auto&& [module_id, struct_id] = *struct_type;
```

Comment:
Please use `const auto&` instead of `auto&&` wherever possible (more places below).